tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,524 @@
+    sB7oXDO8p/+mGL+h7cxFY7hILXTa5/3TGBEgcA65Rrmq22eiRt97RGh\
+    DjfzIqTqb8gwuhTSNN7FWDLrEyRwJMbaTgDSoMIZdLtndVrGEqFHUO+\
+    WzinSiEQCs2MDDnTk29bleHAEktu1x68GYhg9S7O/gZq8/swAV",
+                ),
+                (
+                    "valid2",
+                    "\
+    command=\"echo \\\"test\\\"\" ssh-dss AAAAB3NzaC1kc3MAAACBAJA94Sqw80BSKjVTNZD6570nXIN\
+    hP8R2UhbBuydT+GI6CfA9Dw7O0udJQUfrqARFcRQR/syc72CO6jaKNE\
+    3/A5E+8uVmRZt7s9VtA47s1qxqHswth74m1Nb86n2OTB0HcW63FsXo2\
+    cJF+r+l6F3IcRPi4z/eaEKG7uhAS59TjH2tAAAAFQC0I9kL3oceMT1O\
+    44WPe6NZ8w8CMwAAAIABGm2Yg8nGFZbo/W8njuM79w0W2P1NBVNWzBH\
+    WQqVbr4i1bWTSSc9X+itQUpeF6zAUDsUoprhNise2NLrMYCLFo9JxhE\
+    iYAcEJ/YbKEnjtJzaAmQNpyh3rCWuOcGPTevjAZIkl+zEc+/N7tCW1e\
+    uDYm6IXZ8LEQyTUQUdU4pZ2OgAAAIABk1ZA3+TiCMaoAafNVUZ7zwqk\
+    888yVOgsJ7HGGDGRMo5ytr2SUJB7QWsLX6Un/Zbu32nXsAqtqagxd6F\
+    Ies98TSekMh/hAv9uK92mEsXSINXOeIMKRedqOyPgk5IEOsFpxAUO4T\
+    xpYToeuM8HRemecxw2eIFHnax+mQqCsi7FgQ== core@valid2",
+                ),
+                (
+                    "valid3",
+                    "\
+    sk-ssh-ed25519@openssh.com AAAAGnNrLXNzaC1lZDI1NTE5QG9w\
+    ZW5zc2guY29tAAAAIEX/dQ0v4127bEo8eeG1EV0ApO2lWbSnN6RWusn\
+    /NjqIAAAABHNzaDo= demos@siril",
+                ),
+                (
+                    "bad",
+                    "ssh-bad this-not-a-key core@bad",
+                ),
+            ].iter().cloned().collect();
+
+        static ref FINGERPRINTS: HashMap<&'static str, &'static str> =
+            [
+                (
+                    "valid1",
+                    "SHA256:yZ+o48h6quk9c+JVgJ/Zq4S5u4LUk6TSpneHKkmM9KY",
+                ),
+                (
+                    "valid2",
+                    "SHA256:RP5k1AybZ1kollIAnpUavr1v1nfZ0yloKvI46AMDPkM",
+                ),
+                (
+                    "valid3",
+                    "SHA256:U8IKRkIHed6vFMTflwweA3HhIf2DWgZ8EFTm9fgwOUk",
+                ),
+            ].iter().cloned().collect();
+    }
+
+    // TestContext holds path to a temporary directory used by the current test.
+    struct TestContext {
+        ssh_dir: PathBuf,
+    }
+
+    // Automatically clean up ssh_dir when each test finished.
+    impl Drop for TestContext {
+        fn drop(&mut self) {
+            _ = fs::remove_dir_all(PathBuf::from(self.ssh_dir.clone()));
+        }
+    }
+
+    fn assert_has_keys(ssh_dir: &str, keys: Vec<&str>, assert_no_key: bool) {
+        let authkeyspath: PathBuf = PathBuf::from(format!("{}/authorized_keys", ssh_dir));
+        let authkeysfile = File::open(authkeyspath.clone());
+        assert!(authkeysfile.is_ok());
+        let mut authkeystext = String::new();
+        authkeysfile.unwrap().read_to_string(&mut authkeystext).expect(format!("unable to read a file {}.", authkeyspath.to_str().unwrap_or_default()).as_str());
+        assert!(authkeystext.starts_with("# auto-generated"));
+
+        for key in &keys {
+            if assert_no_key {
```

Comment:
Isn't this switch or the naming switched?

Reply:
renamed to `assert_key_included`.